libbeat/processors/conditionals_test.go

```diff
@@ -122,3 +122,92 @@ func TestConditionRuleInitErrorPropagates(t *testing.T) {
 	assert.Equal(t, testErr, err)
 	assert.Nil(t, filter)
 }
+
+type testCase struct {
+	event common.MapStr
+	want  common.MapStr
+	cfg   string
+}
+
+func testProcessors(t *testing.T, cases map[string]testCase) {
+	for name, test := range cases {
+		test := test
+		t.Run(name, func(t *testing.T) {
+			c, err := common.NewConfigWithYAML([]byte(test.cfg), "test "+name)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			var pluginConfig PluginConfig
+			if err = c.Unpack(&pluginConfig); err != nil {
+				t.Fatal(err)
+			}
+
+			processor, err := New(pluginConfig)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			result, err := processor.Run(&beat.Event{Fields: test.event.Clone()})
+			if err != nil {
+				t.Fatal(err)
+			}
+			assert.Equal(t, test.want, result.Fields)
+		})
+	}
+}
+
+func TestIfElseThenProcessor(t *testing.T) {
+	const ifThen = `
+- if:
+    range.uid.lt: 500
+  then:
+    - add_fields: {target: "", fields: {uid_type: reserved}}
+`
+
+	const ifThenElse = `
+- if:
+    range.uid.lt: 500
+  then:
+    - add_fields: {target: "", fields: {uid_type: reserved}}
+  else:
+    - add_fields: {target: "", fields: {uid_type: user}}
+`
+
+	const ifThenElseSingleProcessor = `
+- if:
+    range.uid.lt: 500
+  then:
+    add_fields: {target: "", fields: {uid_type: reserved}}
+  else:
+    add_fields: {target: "", fields: {uid_type: user}}
+`
```

Comment:
Nothing prevent us to embed a `ifElseProcessor` inside another `IfElseProcessor` maybe we should cover at least a 2 level depth test.

Reply:
This works. I added a test case with a nested `if`.